.github/workflows/gradle.yml

```
@@ -1,68 +1,35 @@
 name: Java CI
 
-on:
-  push:
-    branches:
-      - master
-  pull_request:
-    branches:
-      - master
+on: [push]
 
 jobs:
   build:
+
     runs-on: ubuntu-latest
+
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-          
       - name: Get Fetch Tags
-        if: ${{ !startsWith(github.ref, 'refs/tags') }}
-        run: git fetch --tags --force
-        
+        run: git -c protocol.version=2 fetch --tags --progress --no-recurse-submodules origin
+        if: "!contains(github.ref, 'refs/tags')"
       - name: Set up JDK 11
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           java-version: '11'
-          distribution: 'temurin'
-          
-      - name: Cache Gradle and Assets
-        uses: actions/cache@v3
-        with:
-          path: |
-            ~/.gradle/caches
-            ~/.gradle/wrapper
-            .gradle
-            build/assets
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties', '.gradle/dependency-locks/*') }}
-          restore-keys: |
-            ${{ runner.os }}-gradle-
-            
+          distribution: 'zulu'
       - name: Grant execute permission for gradlew
         run: chmod +x gradlew
-        
-      - name: Build Plugin
-        run: |
-          ./gradlew clean build --info
-        env:
-          GRADLE_OPTS: "-Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2"
-          
+      - name: Build with Gradle
+        run: ./gradlew build
       - name: Get Release Version
         id: get_version
-        run: |
-          VERSION=$(./gradlew currentVersion -q -Prelease.quiet | tail -n 1)
-          echo "VERSION=${VERSION}" >> $GITHUB_OUTPUT
-          echo "Detected version: ${VERSION}"
-          
-      - name: Show build contents
-        run: |
-          echo "Build directory contents:"
-          find build/ -ls
-          
-      - name: Upload Plugin Artifact
+        run: VERSION=$(./gradlew currentVersion -q -Prelease.quiet) && echo ::set-output name=VERSION::$VERSION
```

github will warn about deprecated way to set output
```suggestion
        run: VERSION=$(./gradlew currentVersion -q -Prelease.quiet) && echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
```